wandb/util.py

```diff
@@ -1848,10 +1850,42 @@ def get_core_path() -> str:
     return core_path
 
 
+def working_set() -> Iterable:
+    """Return the working set of installed distributions.
+
+    This function returns a list of namedtuples representing the installed distributions.
+    Each namedtuple has two fields: `key` and `version`. If the `importlib.metadata` module
+    is available, it uses that to get the working set. Otherwise, it falls back to the
+    `importlib_metadata` module.
+    """
+    try:
+        from importlib.metadata import distributions
+    except ImportError:  # Python 3.7
+        from importlib_metadata import distributions  # type: ignore
+    import collections
+
+    WorkingSet = collections.namedtuple("WorkingSet", "key version")
```

Comment:
optional: a dataclass would be nicer here because it can then be referenced in the type

```py
import dataclasses

@dataclasses.dataclass(frozen=True)
class InstalledDistribution:
    """An installed distribution.

    Attributes:
        key: The distribution name, like ...
        version: The distribution's version string.
    """
    key: str
    version: str

def working_set() -> Iterable[InstalledDistribution]:
    ...
```

as an added bonus, putting the docstring on the type allows you to omit that info from the function's docstring

> Return the working set of installed distributions.
>
> Uses `importlib.metadata` in Python versions above 3.7, and `importlib_metadata` otherwise.

Reply:
sounds good... i'll redo it